File: src/main/java/org/parham/seasonjob/commands/default_cmd/Deny.java
package org.parham.seasonjob.commands.default_cmd;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.parham.seasonjob.commands.SeasonCommand;
import org.parham.seasonjob.data.member.MemberManager;
import org.parham.seasonjob.data.sender.Messages;

import java.util.ArrayList;
import java.util.List;

public class Deny implements SeasonCommand {
    @Override
    public boolean hasPermission() {
        return false;
    }

    @Override
    public boolean isPlayerOnly() {
        return true;
    }

    @Override
    public String getPermission() {
        return "seasonjob.deny";
    }

    @Override
    public int getNeedArguments() {
        return 0;
    }

    @Override
    public String getUsage() {
        return "";
    }

    @Override
    public void execute(CommandSender sender, Command cmd, String label, String[] args, boolean isLeader, boolean hasAccess) {
        Player player = (Player) sender;

        if (!Invite.invites.containsKey(player.getUniqueId())) {
            player.sendMessage(Messages.getMessage("doesnt-have-invite-deny"));
        } else {
            Invite.invites.remove(player.getUniqueId());
            player.sendMessage(Messages.getMessage("denied-deny"));
        }
    }

    @Override
    public List<String> getCompletions(CommandSender sender, String[] args) {
        return new ArrayList<>();
    }
}

File: src/main/kotlin/ir/parham/seasonJob/Commands/Default/Deny.kt
package ir.parham.seasonJob.Commands.Default

import Libs.API.ir.parham.SeasonJobsAPI.Commands
import Libs.API.ir.parham.SeasonJobsAPI.Senders.Message
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*
import kotlin.collections.List

class Deny : Commands {
    override fun runner(sender: CommandSender, args: Array<out String>)
    {
        val message = Message()

        if (sender !is Player) {
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "denyConsole"))
            // just player
        } else if (!sender.hasPermission("seasonjobs.deny")) {
            sender.sendMessage(message.get(sender, "denyPerm"))
            // deny perm
        } else if (!Invite.invite.containsKey(sender.uniqueId)) {
            sender.sendMessage(message.get(sender, "emptyInvite"))
            // no invite
        } else {
            remover(sender)
            sender.sendMessage(message.get(sender, "denySuccess"))
            // deny succeed
        }
    }

    override fun completer(sender: CommandSender, args: Array<out String>): List<String>
    {
        return listOf("")
    }

    override fun remover(target: OfflinePlayer)
    {
        Invite.invite.remove(target.uniqueId)
    }

    override fun adder(target: OfflinePlayer)
    {
        TODO("Not yet implemented")
    }

    override fun setter(target: OfflinePlayer)
    {
        TODO("Not yet implemented")
    }
}